lib/parsers/iso_base_media_file_format/utils.rb

```diff
@@ -62,9 +62,9 @@ def frame_rate(box_tree)
       end
 
       def video_codecs(box_tree)
-        video_trak_boxes(box_tree).map do |trak_box|
-          trak_box.all_descendents_by_path(%w[mdia minf stbl stsd]).map { |stsd_box| stsd_box.children.map(&:type) }
-        end.flatten.compact.uniq
+        video_trak_boxes(box_tree).flat_map do |trak_box|
+          trak_box.all_descendents_by_path(%w[mdia minf stbl stsd]).flat_map { |stsd_box| stsd_box.children.map(&:type) }
```

Comment:
Do we need this nested `flat_map` as well? Looking at the diff, it's different than before.

Reply:
We either need a nested `flat_map` _or_ we need to revery to the `map` and `flatten` on the outer block, because `flat_map` != `map` & `flatten`.

`flat_map` will only flatten to a depth of 1 and will not totally flatten arrays of arrays.